static/app/views/alerts/rules/issue/setupMessagingIntegrationButton.tsx

```diff
@@ -0,0 +1,109 @@
+import styled from '@emotion/styled';
+
+import {openModal} from 'sentry/actionCreators/modal';
+import {Button} from 'sentry/components/button';
+import {Tooltip} from 'sentry/components/tooltip';
+import {t} from 'sentry/locale';
+import PluginIcon from 'sentry/plugins/components/pluginIcon';
+import {space} from 'sentry/styles/space';
+import type {Project} from 'sentry/types/project';
+import {trackAnalytics} from 'sentry/utils/analytics';
+import {useApiQuery} from 'sentry/utils/queryClient';
+import useRouteAnalyticsParams from 'sentry/utils/routeAnalytics/useRouteAnalyticsParams';
+import useOrganization from 'sentry/utils/useOrganization';
+import MessagingIntegrationModal from 'sentry/views/alerts/rules/issue/messagingIntegrationModal';
+
+interface ProjectWithAlertIntegrationInfo extends Project {
+  hasAlertIntegrationInstalled: boolean;
+}
+
+type Props = {
+  projectSlug: string;
+  refetchConfigs: () => void;
+};
+
+function SetupMessagingIntegrationButton({projectSlug, refetchConfigs}: Props) {
+  const providerKeys = ['slack', 'discord', 'msteams'];
+  const organization = useOrganization();
+
+  const onAddIntegration = () => {
+    refetch();
+    refetchConfigs();
+  };
+
+  const {
+    data: project,
+    isLoading,
+    isError,
+    refetch,
+  } = useApiQuery<ProjectWithAlertIntegrationInfo>(
+    [
+      `/projects/${organization.slug}/${projectSlug}/`,
+      {query: {expand: 'hasAlertIntegration'}},
+    ],
+    {staleTime: Infinity}
+  );
+
+  const shouldRenderSetupButton = project && !project.hasAlertIntegrationInstalled;
+
+  useRouteAnalyticsParams({
+    setup_message_integration_button_shown: shouldRenderSetupButton,
+  });
+
+  if (isLoading || isError) {
+    return null;
+  }
+
+  if (!shouldRenderSetupButton) {
+    return null;
+  }
+
+  // TODO(Mia): only render if organization has team plan and above
```

Comment:
just want to make sure it's okay to merge w/ this todo.

Reply:
Yup! That will be in my next PR, it's fine for now since this button is hidden behind a feature flag :)